packages/lib/src/utils/observability/honeycomb.ts

```diff
@@ -0,0 +1,96 @@
+import { HoneycombWebSDK } from "@honeycombio/opentelemetry-web";
+import { getWebAutoInstrumentations } from "@opentelemetry/auto-instrumentations-web";
+import { detectGraphqlQuery } from "./utils";
+
+/**
+ * Configuration object for the Honeycomb SDK.
+ */
+interface HoneycombConfig {
+  /** The service name for the app we are running e.g. spruce or parsley */
+  serviceName: string;
+  /** The endpoint for the Honeycomb SDK to send traces to if we are not using the default */
+  endpoint: string;
+  /** The url for our Honeycomb instrumented server to connect frontend and backend traces together */
+  backendURL?: string;
+  /** Whether to enable debug mode in the Honeycomb SDK this will enable additional logging and print links to traces */
+  debug: boolean;
+  /** The INGEST key for the Honeycomb SDK */
+  ingestKey: string;
+}
+
+/**
+ * Initializes the Honeycomb SDK with the provided configuration.
+ * @param config - The configuration object for the Honeycomb SDK.
+ * @param config.ingestKey - The Honeycomb INGEST API key.
+ * @param config.backendURL - The backend URL.
+ * @param config.debug - Whether to start the SDK in debug mode.
+ * @param config.serviceName - The name of the service.
+ * @param config.endpoint - The endpoint for the Honeycomb SDK to send traces to if we are not using the default.
+ */
+const initializeHoneycomb = ({
+  backendURL,
+  debug,
+  endpoint,
+  ingestKey,
+  serviceName,
+}: HoneycombConfig) => {
+  if (debug && !ingestKey) {
+    console.warn(
+      "Honeycomb INGEST API key was not provided. Starting SDK in debug mode.",
+    );
+    return;
+  }
+  if (!ingestKey || !endpoint) {
+    console.error(
+      "Honeycomb INGEST API key or a collector endpoint is required to start the SDK in production mode.",
+    );
+  } else {
+    try {
+      const userId = localStorage.getItem("userId") ?? undefined;
+      const honeycombSdk = new HoneycombWebSDK({
+        debug,
+        endpoint,
```

Comment:
how exactly does this work if both the `endpoint` and `apiKey` are defined? Don't we by default want to send to the `endpoint` and ignore the `apiKey`? If a collector `endpoint` is provided, it seems like we shouldn't need to provide an API key

<img width="567" alt="Screenshot 2024-09-10 at 12 32 03 PM" src="https://github.com/user-attachments/assets/dc4a4a48-bd83-476f-8a13-3bac1fcc129c">

Reply:
It will always send to the collector if the collector endpoint is supplied. The apiKey will just be ignored.

Reply:
in that case yippee!